docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
+//
+// +optional
+JoinSubnets DualStackCIDRs `json:"joinSubnets,omitempty"`
+
++ // infrastructureSubnets specifies a list of internal CIDR ranges that OVN-Kubernetes will reserve for internal network infrastructure.
++ // Any IP addresses within these ranges cannot be assigned to workloads.
++ // When omitted, OVN-Kubernetes will automatically allocate IP addresses from `subnets` for its infrastructure needs.
++ // When `excludeSubnets` is also specified the CIDRs cannot overlap.
++ // When `defaultGatewayIPs` is also specified  the default gateway IPs cannot be a part of any of the CIDRs.
++ // Each item should be in range of the specified CIDR(s) in `subnets`.
++ // The maximal exceptions allowed is 10.
++ // The format should match standard CIDR notation (for example, "10.128.0.0/16").
++ // This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
++ // +optional
++ // +kubebuilder:validation:MinItems=1
++ // +kubebuilder:validation:MaxItems=10
++ InfrastructureSubnets []CIDR `json:"infrastructureSubnets,omitempty"`
+
++ // defaultGatewayIPs specifies the default gateway IP used in the internal OVN topology.
++ //
++ // Dual-stack clusters may set 2 IPs (one for each IP family), otherwise only 1 IP is allowed.
++ // This field is only allowed for "Primary" network.
++ // It is not recommended to set this field without explicit need and understanding of the OVN network topology.
++ // When omitted, the first IP address from the pod network subnet is used.
++ //
++ // +optional
++ DefaultGatewayIPs DualStackIPs `json:"defaultGatewayIPs,omitempty"`
+
+// IPAM section contains IPAM-related configuration for the network.
+// +optional
+IPAM *IPAMConfig `json:"ipam,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="isIP(self)", message="IP is invalid"
+type IP string
+
+// +kubebuilder:validation:MinItems=1
+// +kubebuilder:validation:MaxItems=2
+// +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isIP(self[0]) || !isIP(self[1]) || ip(self[0]).family() != ip(self[1]).family()", message="When 2 IPs are set, they must be from different IP families"
+type DualStackIPs []IP
+
+```
+
+#### IPAMClaim API changes
+
+The following pull request is tracking the IPAMClaim API change that introduces the status conditions:
+https://github.com/k8snetworkplumbingwg/ipamclaims/pull/9
+
+[IPAMClaim CRD doc](https://docs.google.com/document/d/1OQIJIrCtsYpR5O44w0hpoJ2TyKBz1Du-KhRT4RtrAjk) - `IPAM allocation on behalf of other entities` section
+
+### Implementation Details
+
+#### Configurability
+
+The changes outlined in this enhancement should be configurable. This means a configuration knob
+is required to instruct OVN-Kubernetes on whether to process the annotation described in the 
+[Pod network identity](#pod-network-identity) section.
+
+#### NetworkSelectionElement annotation
+
+Currently, the `v1.multus-cni.io/default-network` annotation is only processed for the cluster default network.
```

Comment:
I think it would also be good to call out either here or somwhere else that the value that this annotation takes is a NetworkSelectionElement.